internal/operator/operator.go

```diff
@@ -457,20 +451,43 @@ func applyCertDiscoveryVenafiManifests(mf *manifests, options ApplyInstallationY
 		return
 	}
 	cdv, secret := venafi.GenerateManifestsForCertDiscoveryVenafi(options.CertDiscoveryVenafi)
-	var imagePullSecrets []string
-	if options.RegistryCredentialsPath != "" || options.RegistryCredentials != "" {
-		imagePullSecrets = []string{"jse-gcr-creds"}
-	}
-	// Eventually we probably want to have a single field for image pull
-	// secrets on Installation resource, but this change will happen in the
-	// operator.
-	if cdv != nil {
-		cdv.ImagePullSecrets = imagePullSecrets
-	}
 	mf.installation.Spec.CertDiscoveryVenafi = cdv
 	mf.secrets = append(mf.secrets, secret)
 }
 
+func applyImagePullSecretToManifests(mf *manifests, options ApplyInstallationYAMLOptions) error {
+	registryCredentials := options.RegistryCredentials
+	if registryCredentials == "" && options.RegistryCredentialsPath != "" {
+		registryCredentialsBytes, err := os.ReadFile(options.RegistryCredentialsPath)
+		if err != nil {
+			return fmt.Errorf("failed to read registry credentials file: %w", err)
+		}
+		registryCredentials = string(registryCredentialsBytes)
+	}
+
+	if registryCredentials != "" {
+		secret, err := registry.ImagePullSecret(registryCredentials)
+		if err != nil {
+			return fmt.Errorf("failed to parse image pull secret: %w", err)
+		}
+		mf.secrets = append(mf.secrets, secret)
+		if mf.installation.Spec.Images == nil {
+			mf.installation.Spec.Images = &operatorv1alpha1.Images{}
+		}
+		mf.installation.Spec.Images.Secret = secret.Name
```

Comment:
Can you explain why we are altering the installation manifest here? Is this change updated in the cluster state too? How does this work?

Reply:
We are actually building the installation manifest here, thank you @irbekrm for explaining this.